.circleci/config.yml

```diff
@@ -17,6 +17,9 @@ jobs:
         command: |
           python3.7 -m venv venv/
           venv/bin/pip install --upgrade -r requirements.txt
+    - run:
+        name: Generate SQL queries with UDFs
+        command: python3.7 script/generate_sql.py
```

Comment:
We should make it so we don't needing to call `script/generate_sql` before running pytest during development

Option A:

1. add a `script/build` like
    ```bash
    #!/usr/bin/env bash
    cd "$(dirname "$0")"
    if [ "$#" != 0 ]; then
        echo "usage: script/$(basename "$0")" >&2
        echo "unrecognized arguments: $*" >&2
        exit 1
    fi
    script/generate_sql
    ```
2. add a `script/test` like
    ```bash
    #!/usr/bin/env bash
    cd "$(dirname "$0")"
    script/build
    exec pytest "$@"
    ``` 
3. modify `script/entrypoint` to call `script/test` instead of `pytest`
4. add a step after `PyTest with linters` named `Verify that all generated files are committed` that runs `git diff --exit-code target/`

Option B (preferred):

Use a makefile to the same effect as option A

Reply:
@jklukas i'm interested in your opinion on makefile vs other build systems, to make sure we build files as-needed.

Reply:
I agree that a Makefile is probably going to be needed long-term.

For the scope of this PR, however, I'd like to see us factor out this sql parsing code to a library that can be called by tests and as a script.

We could take the existing content of `tests/parse_udf.py` and move that instead into a new `src` directory or `bigquery_etl` directory that will serve as a python module. If we have a `bigquery_etl` directory, then I believe `script/generate_sql` will be able to `import bigquery_etl` without having to modify `sys.path`.

Reply:
I added a `Makefile`, however I'm not sure whether it should be called in `entrypoint` or somewhere else?

I also think that `parse_udf` should be moved out of `tests`. I was however not able to make it usable as a python module just by simply calling `import bigquery_etl`. The problem I ran into is described in https://stackoverflow.com/questions/6323860/sibling-package-imports/23542795#23542795 there is also a solution using setuptools that does not require to modify `sys.path` but it felt more like a hack (it required to add a `setup.py`, running pip install on the project etc.). I decided to stay with modifying `sys.path` for now but am open to better suggestions.